### PR TITLE
Proxy Port Improvements

### PR DESCRIPTION
**Pull request by marek-piirikivi**
I added additional check to see if there exists an item on the diagram that may own the proxy port.

### PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have read, and I am following the [Contributor guide](https://github.com/gaphor/gaphor/blob/main/CONTRIBUTING.md)
- [x] I have read, and I understand the GNOME [Code of Conduct](https://wiki.gnome.org/Foundation/CodeOfConduct)

### PR Type
What kind of change does this PR introduce?

- [x] Bug fix
- [ ] Feature
- [ ] Chore (refactoring, formatting, local variables, other cleanup)
- [ ] Documentation content changes

### What is the current behavior?

Issue Number: #2612

### Does this PR introduce a breaking change?
- [ ] Yes
- [x] No

### Other information

Additionally fixed proxy port drag icon. Automatically generated icon name is "proxy-port".

### CODE REVIEW

File: gaphor/SysML/drop.py [hunk 1]
```diff
@@ -11,7 +13,23 @@ def drop_proxy_port(element: ProxyPort, diagram, x, y):
...
+            isinstance(item, PropertyItem)
+            and isinstance(item.subject.type, Block)
+            and element in item.subject.type.ownedPort
+        )
+
+    head_item = next(
+        (
+            item
+            for item in diagram.select(
+                property_item_which_is_typed_by_block_owning_the_port
```
Comment (amolenaar) [inline gaphor/SysML/drop.py:27]: I would inline the code.

```suggestion
                lambda e: isinstance(e, PropertyItem)
                and isinstance(e.subject.type, Block)
                and element in e.subject.type.ownedPort
```

**Comment by danyeaw** [comment]
Hi @marek-piirikivi, I tried to test this by doing the following:

1. Create a block called Car
2. Create a 2nd block named Wheel
3. Add a composition relationship from the Car to the Wheel
4. Add the Wheel Part Property to the diagram
5. Add a proxy port to the Wheel Part Property
6. Add the Wheel Part Property to the diagram again
7. Try to drag the proxy port on to the 2nd Wheel Part Property

However, this doesn't seem to work for me. Am I testing this correctly or was it fixing some other use case?

**Comment by marek-piirikivi** [comment]
@danyeaw Could you tell me what symptoms there were? What happened? What you expected to happen?

I'll check this out myself, but the fix is about reusing already existing elements from the model browser. It does not fix the problem that we have before about the problem with placing ports when there are multiple items that can hold the port on the diagram. It is known older problem that when there are multiple items which the port can belong to, that only the first found item gets the port - is this what you experienced? I can solve this within this merge request as well, but I see this as a separate problem. This change merely "adds block's property into the list that even can own a port that's dragged from the model btowser". :slightly_smiling_face:

**Comment by danyeaw** [comment]
@marek-piirikivi Sorry for the confusion, it sounds like I was testing out a different use case, definitely don't expand this PR to fix that other thing. I reread the issue you linked to. With this PR, if I try this:

1. Create a block
2. Use the corresponding part on an ibd
3. Add a port to the part
4. Go to the bdd, and double click the port in the model browser

The port is not adding to the part again instead of the block.

**Comment by marek-piirikivi** [comment]
@danyeaw Are you sure you are looking at the correct changes? Here is a video - I tried to reproduce the issues, but maybe I am not understanding the use-case properly :thinking: 

https://github.com/gaphor/gaphor/assets/45357878/86d4414a-69c0-4d58-9b9f-28becfcd4978

**Comment by marek-piirikivi** [comment]
> definitely don't expand this PR to fix that other thing

:smiling_imp: I did that. It was bugging me so much... even if I do not see the use-case for having the two items representing the same block on the same diagram.

Removing unused elements with proxy port is still broken. When I delete it from any of the items referencing it, it removes it from everywhere.

**Comment by marek-piirikivi** [comment]
... I should add tests as well... :sweat_smile:

**Comment by marek-piirikivi** [comment]
@danyeaw @amolenaar 

I made some updates:
1. You can now drag existing ports onto all the items that may own the ports - either the owning block or a property typed by owning block. The closest item will get the block.
2. Deleting a port item will not delete the port from the model if there are any port items presenting the port. I tested manually, seems to be working with both "Remove unused elements" enabled and disabled as expected. Now, making a mistake while dragging a duplicate port, you can remove it without having to worry about it being removed from the model.
3. I added tests for the first one. Second one had already test coverage and it did not brake.

From my point of view, I tried to brake the functionality, but couldn't - it behaves as I expect it to behave. Please review. :slightly_smiling_face: 

Here is a small demo as well (ignore jerky mouse movement - I am not very good with mouse, shaky hands)
https://github.com/gaphor/gaphor/assets/45357878/a77847ab-1f4a-4c91-acca-0763dcd9d6bf

**Review by danyeaw — APPROVED** [review]
Thanks @marek-piirikivi!

**Pull request merged**